contracts/src/assertionStakingPool/EdgeStakingPool.sol

```diff
@@ -10,45 +10,50 @@ import "../challengeV2/EdgeChallengeManager.sol";
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
-/// @notice Staking pool contract for creating layer zero edges.
-/// Allows users to deposit stake, create an edge once required stake amount is reached,
-/// and reclaim their stake when and if the edge is confirmed.
-/// @dev Unlike the assertion staking pool, there is no need for a function to claim the stake back into the pool.
+/// @notice Trustless staking pool contract for creating layer zero edges.
+///
+///         Allows users to deposit stake, create an edge once required stake amount is reached,
+///         and reclaim their stake when and if the edge is confirmed.
+///
+///         Honest participants should check that both edgeId and edgeLevel are correct before staking. 
+///         If either is not expected, the pool should be ignored.
+///
+/// @dev    Unlike the assertion staking pool, there is no need for a function to claim the stake back into the pool.
 contract EdgeStakingPool is AbsBoldStakingPool {
     using SafeERC20 for IERC20;
 
     /// @notice The targeted challenge manager contract
     EdgeChallengeManager public immutable challengeManager;
-    /// @notice todo
-    bytes32 public immutable createEdgeArgsHash;
+    /// @notice The ID of the edge to be created (see ChallengeEdgeLib.id)
+    bytes32 public immutable edgeId;
+    /// @notice The level of the edge to be created
+    uint8 public immutable edgeLevel;
     /// @notice The required stake amount to create the edge
     uint256 public immutable requiredStake;
 
     /// @notice The provided arguments to not match createEdgeArgsHash
-    error IncorrectCreateEdgeArgs();
+    error IncorrectEdgeId(bytes32 actual, bytes32 expected);
 
     /// @param _challengeManager EdgeChallengeManager contract
-    /// @param createEdgeArgs Arguments to be passed into EdgeChallengeManager.createLayerZeroEdge
+    /// @param _edgeId The ID of the edge to be created (see ChallengeEdgeLib.id)
+    /// @param _edgeLevel The level of the edge to be created
     constructor(
         address _challengeManager,
-        CreateEdgeArgs memory createEdgeArgs
+        bytes32 _edgeId,
+        uint8 _edgeLevel
```

Comment:
I don't think it is possible to have the same edgeId for different level, so we can omit this.